frontend/src/metabase/api/utils/use-token-refresh.unit.spec.tsx

```diff
@@ -0,0 +1,95 @@
+import { setupPropertiesEndpoints } from "__support__/server-mocks";
+import { act, renderWithProviders, screen, waitFor } from "__support__/ui";
+import { findRequests } from "__support__/utils";
+import { createMockSettings } from "metabase-types/api/mocks";
+
+import { useGetSettingsQuery } from "../session";
+
+import { useTokenRefresh } from "./use-token-refresh";
+
+const TestComponent = () => {
+  useTokenRefresh();
+  const { isFetching } = useGetSettingsQuery();
+
+  if (isFetching) {
+    return <div>Loading...</div>;
+  }
+
+  return <div>Test</div>;
+};
+
+const waitTime = async () => {
+  act(() => {
+    jest.advanceTimersByTime(11 * 1000);
+  });
+  await screen.findByText("Test");
+};
+
+const setupRefreshableProperties = (hasRefresh = true) => {
+  const settings = createMockSettings({
+    "site-name": "Test",
+    "token-status": {
+      valid: true,
+      status: "Token is Valid.",
+      features: hasRefresh ? ["refresh-token-features"] : [],
+    },
+  });
+  setupPropertiesEndpoints(settings);
+};
+
+const setup = async (hasRefresh = true) => {
+  setupRefreshableProperties(hasRefresh);
+  renderWithProviders(<TestComponent />);
+  await screen.findByText("Loading...");
+  await screen.findByText("Test");
+  return waitForGets(1);
+};
+
+const waitForGets = async (count: number) => {
+  return waitFor(async () => {
+    const gets = await findRequests("GET");
+    expect(gets.length).toBe(count);
+  });
+};
+
+describe("useTokenRefresh", () => {
+  beforeAll(() => {
+    jest.useFakeTimers({ advanceTimers: true });
+  });
+
+  it("should refetch every 10 seconds", async () => {
+    await setup(true); // always start with 1 request
+    await waitTime(); // 2
+    expect((await findRequests("GET")).length).toBe(2);
+    await waitTime(); // 3
+    await waitTime(); // 4
+    await waitForGets(4);
```

Comment:
Someone might get confused because 'wait' has quite different meanings on lines 65 and 66. That's one reason in favor of using the name "advanceTimersByElevenSeconds" instead of "waitTime"